src/overlays/actors/ovl_En_Mag/z_en_mag.c

```diff
@@ -27,21 +33,553 @@ const ActorInit En_Mag_InitVars = {
+
+    func_8006EF10(&this->unk_150);
+
+    this->unk_E316 = 0;
+    this->unk_E318 = 0;
+    this->unk_E31C = 0;
+    this->unk_E320 = 0;
+}
+
+void EnMag_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void EnMag_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad[2];
+    EnMag* this = THIS;
+
+    if (gSaveContext.fileNum != 0xFEDC) {
+        if (this->globalState < MAG_STATE_DISPLAY) {
+            if (CHECK_PAD(globalCtx->state.input[0].press, START_BUTTON) ||
+                CHECK_PAD(globalCtx->state.input[0].press, A_BUTTON) ||
+                CHECK_PAD(globalCtx->state.input[0].press, B_BUTTON)) {
+
+                Audio_PlaySoundGeneral(NA_SE_SY_PIECE_OF_HEART, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+
+                this->mainAlpha = 210;
+                this->subAlpha = 255;
+                this->copyrightAlpha = 255;
+
+                this->effectPrimLodFrac = 128.0f;
+                this->effectAlpha = 255.0f;
+
+                this->effectPrimColor[0] = 170;
+                this->effectPrimColor[1] = 255.0f;
+                this->effectPrimColor[2] = 255.0f;
+                this->effectEnvColor[0] = 200.0f;
+                this->effectEnvColor[1] = 255.0f;
+                this->effectEnvColor[2] = 0;
+
+                this->globalState = MAG_STATE_DISPLAY;
+                sDelayTimer = 20;
+                gSaveContext.fadeDuration = 1;
+                gSaveContext.unk_1419 = 255;
+            }
+        } else if (this->globalState >= MAG_STATE_DISPLAY) {
+            if (sDelayTimer == 0) {
+                if (CHECK_PAD(globalCtx->state.input[0].press, START_BUTTON) ||
+                    CHECK_PAD(globalCtx->state.input[0].press, A_BUTTON) ||
+                    CHECK_PAD(globalCtx->state.input[0].press, B_BUTTON)) {
+
+                    if (globalCtx->sceneLoadFlag != 20) {
+                        func_800F68BC(0);
+
+                        Audio_PlaySoundGeneral(NA_SE_SY_PIECE_OF_HEART, &D_801333D4, 4, &D_801333E0, &D_801333E0,
+                                               &D_801333E8);
+
+                        gSaveContext.gameMode = 2;
+                        globalCtx->sceneLoadFlag = 20;
+                        globalCtx->fadeTransition = 2;
+                    }
+
+                    this->copyrightAlphaStep = 15;
+                    this->fadeOutAlphaStep = 25;
+                    this->globalState = MAG_STATE_FADE_OUT;
+                }
+            } else {
+                sDelayTimer--;
+            }
+        }
+    }
+
+    if (this->globalState == MAG_STATE_FADE_IN) {
+        if (this->effectFadeInState == 0) {
+            this->effectAlpha += 6.375f;
+            this->effectPrimLodFrac += 0.8f;
+
+            this->effectPrimColor[0] += 6.375f;
+            this->effectPrimColor[1] += 3.875f;
+            this->effectPrimColor[2] += 2.125f;
+            this->effectEnvColor[0] += 6.375f;
+            this->effectEnvColor[1] += 3.875f;
```

Comment:
I guess that's a "not interested" then lol, no problem

Reply:
I was thinking about it but I feel like this is a bit out of scope for a first documentation pass. I could see us using a pattern like that eventually if it can be consistent throughout the file, but probably only after making defines for the different notable values (such as the timer duration of 40 here).